easybuild/easyblocks/c/clang.py

```diff
@@ -164,27 +177,52 @@ def find_source_dir(globpatterns, targetdir):
                                      glob_src_dirs)
             src_dirs[glob_src_dirs[0]] = targetdir
 
-        find_source_dir('compiler-rt-*', os.path.join(self.llvm_src_dir, 'projects', 'compiler-rt'))
-
-        if self.cfg["usepolly"]:
-            find_source_dir('polly-*', os.path.join(self.llvm_src_dir, 'tools', 'polly'))
-
-        if self.cfg["build_lld"]:
-            find_source_dir('lld-*', os.path.join(self.llvm_src_dir, 'tools', 'lld'))
-            if LooseVersion(self.version) >= LooseVersion('12.0.1'):
-                find_source_dir('libunwind-*', os.path.normpath(os.path.join(self.llvm_src_dir, '..', 'libunwind')))
-
-        if self.cfg["libcxx"]:
-            find_source_dir('libcxx-*', os.path.join(self.llvm_src_dir, 'projects', 'libcxx'))
-            find_source_dir('libcxxabi-*', os.path.join(self.llvm_src_dir, 'projects', 'libcxxabi'))
-
-        find_source_dir(['clang-[1-9]*', 'cfe-*'], os.path.join(self.llvm_src_dir, 'tools', 'clang'))
-
-        if self.cfg["build_extra_clang_tools"]:
-            find_source_dir('clang-tools-extra-*', os.path.join(self.llvm_src_dir, 'tools', 'clang', 'tools', 'extra'))
-
-        if LooseVersion(self.version) >= LooseVersion('3.8'):
-            find_source_dir('openmp-*', os.path.join(self.llvm_src_dir, 'projects', 'openmp'))
+        if LooseVersion(self.version) >= LooseVersion('14'):
+            # the expected directory structure has completely changed in version 14.x of the project, we use the
```

Comment:
The tarballs `llvm-project-14.0.0.src` and `llvm-project-13.0.0.src` don't look that different and LLVM v13 supports `LLVM_ENABLE_PROJECTS` as well.

Instead of checking for version 14+ or not, should we check if the sources of the easyconfig have an `llvm-project-*` tarball, and if so proceed in the new way with `LLVM_ENABLE_PROJECTS`?

Reply:
sure, I can do that as well

Reply:
addressed in ae86d8876